relentless/potential/potential.py

```diff
@@ -294,48 +312,91 @@ def __init__(self, types, params, default={}):
 
         self.coeff = CoefficientMatrix(types, params, default)
 
-        self.id = PairPotential._id
-        PairPotential._id += 1
-
     def energy(self, pair, r):
-        """ Evaluate energy for a (i,j) pair."""
+        """Evaluate energy for a (i,j) pair.
+
+        Parameters
+        ----------
+        pair : array_like
+            The pair for which to calculate the energy.
+        r : array_like
+            The location(s) at which to evaluate the energy.
+
+        Returns
+        -------
+        array_like
+            The energy at the specified location(s).
```

Comment:
You might want to clarify that the returned quantity will have the same form as `r`. So, if r is a scalar, it will be a scalar, but if r is array_like, it will be a numpy array.